gst/nnstreamer/tensor_query/tensor_query_common.c

```diff
@@ -530,7 +530,14 @@ nnstreamer_query_server_data_free (query_server_handle server_data)
       if (sdata->conn_queue) {
         while ((conn_remained = g_async_queue_try_pop (sdata->conn_queue))) {
           conn_remained->running = 0;
-          pthread_join (conn_remained->msg_thread, NULL);
+          if (conn_remained->msg_thread) {
```

Comment:
Suggestion: stopping thread in close function - move this block into nnstreamer_query_close()

Reply:
Yes, that looks better. updated.